src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -560,6 +565,27 @@ ErrorIfNotSupportedForeignConstraint(Relation relation, char distributionMethod,
 			continue;
 		}
 
+		referencedTableId = constraintForm->confrelid;
+		selfReferencingTable = relation->rd_id == referencedTableId;
+
+		/*
+		 * We do not support foreign keys for reference tables. Here we skip the second
+		 * part of check if the table is a self referencing table because;
+		 * - PartitionMethod only works for distributed tables and this table is not
+		 * distributed yet.
+		 * - Since referencing and referenced tables are same, it is OK to not checking
+		 * distribution method twice.
+		 */
+		if (distributionMethod == DISTRIBUTE_BY_NONE ||
+			(!selfReferencingTable &&
```

Comment:
This check is there so that we will not reject hash distributing self referencing tables. Without that check we would check for;

PartitionMethod(referencedTableId) == DISTRIBUTE_BY_NONE

and error out because that table is not distributed yet.